cxx/hirm.cc

```diff
@@ -49,6 +49,23 @@ void HIRM::transition_cluster_assignments(std::mt19937* prng,
   }
 }
 
+void HIRM::add_relation_to_irm(IRM* irm, const std::string& r,
+                               const T_relation& t_relation) {
+  std::visit(
+      [&](const auto& trel) {
+        using T = std::decay_t<decltype(trel)>;
```

Comment:
This project is certainly increasing my knowledge of all the dark corners of <type_traits>.  [No response required].

Reply:
Same! I picked this one up from the example in https://en.cppreference.com/w/cpp/utility/variant/visit